segment_language.go

```diff
@@ -18,6 +32,20 @@ func (l *language) string() string {
 }
 
 func (l *language) enabled() bool {
+	displayMode := DisplayMode(l.props.getString(DisplayModeProperty, string(DisplayModeContext)))
+	displayVersion := l.props.getBool(DisplayVersion, true)
+	hasVersion := l.getVersion()
+
+	if displayMode == DisplayModeAlways {
```

Comment:
I'm a strong proponent of [object calisthenics](https://williamdurand.fr/2013/06/03/object-calisthenics/), and try to avoid `else` most I can. It's better to use a `switch` statement in this case. Taking my previous comment in consideration:

```golang
func (l *language) enabled() bool {
	displayMode := l.props.getString(DisplayModeProperty, string(DisplayModeContext))
	displayVersion := l.props.getBool(DisplayVersion, true)
	hasVersion := l.getVersion()
	switch displayMode {
	case DisplayModeAlways:
		return (hasVersion || !displayVersion)
	case DisplayModeNever:
		return false
	default:
		return l.isInContext() && (hasVersion || !displayVersion)
	}
}
```